### PR TITLE
Added '.' and '..' special folder t WASI fd_readdir return (for #3033)

### PR DESCRIPTION
**Pull request by ptitSeb**
# Description
Simply adding "." and ".." to the returnned list from an WASI::fd_readdir call

### CODE REVIEW

**Comment by syrusakbary** [comment]
This should fix #3033

File: lib/wasi/src/syscalls/mod.rs [hunk 1]
```diff
@@ -1321,6 +1321,10 @@ pub fn fd_readdir<M: MemorySize>(
                             (entry.name.to_string(), stat.st_filetype, stat.st_ino)
                         }),
                 );
+                // adding . and .. special folders
+                // TODO: inode
+                entry_vec.push((".".to_string(), __WASI_FILETYPE_DIRECTORY, 0));
+                entry_vec.push(("..".to_string(), __WASI_FILETYPE_DIRECTORY, 0));
```
Comment (syrusakbary) [inline lib/wasi/src/syscalls/mod.rs:1327]: Ummm... what about the root? We shouldn't add `..` to readdir in the root.
Can you also add a test case for this? Thanks

Comment (ptitSeb) [reply lib/wasi/src/syscalls/mod.rs:1327]: But, isn't Root the next `match` here?

Comment (syrusakbary) [reply lib/wasi/src/syscalls/mod.rs:1327]: Also, side comment. Do you have insight on why our fix is quite different from wasmtime's one?

https://github.com/bytecodealliance/wasmtime/pull/2494/files#diff-df2ef1370349b8385454adfabfe871fc51fd3c12debc7b0691b98ca19c5724deR309-R335

Comment (ptitSeb) [reply lib/wasi/src/syscalls/mod.rs:1327]: What I hesitated is adding "." to a root readdir

Comment (ptitSeb) [reply lib/wasi/src/syscalls/mod.rs:1327]: the root cause was completly different: it's not a buffer size issue here, it's the missing "." and "..". All the 201 folder appears correctly, just missing the special folders

File: lib/wasi/src/syscalls/mod.rs [hunk 2]
```diff
@@ -1321,6 +1321,10 @@ pub fn fd_readdir<M: MemorySize>(
                             (entry.name.to_string(), stat.st_filetype, stat.st_ino)
                         }),
                 );
+                // adding . and .. special folders
+                // TODO: inode
```
Comment (syrusakbary) [inline lib/wasi/src/syscalls/mod.rs:1325]: What is this TODO?

Comment (ptitSeb) [reply lib/wasi/src/syscalls/mod.rs:1325]: same as line 1310

File: lib/wasi/src/syscalls/mod.rs (hunk 1 above)
Comment (ptitSeb) [reply lib/wasi/src/syscalls/mod.rs:1327]: So no, adding "." to root is a bad idea, it breaks things.

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**